ui/views.py

```diff
@@ -24,3 +26,48 @@ def get(self, request, media_key):
         response = super().get(request, media_key)
         response.data['media_item_json'] = json.dumps(response.data)
         return response
+
+
+class MediaAnalyticsView(api.views.MediaAnalyticsView):
+    """
+    View for rendering an individual media item's analytics.
+    Extends the DRF's media item analytics view.
+
+    """
+    renderer_classes = [TemplateHTMLRenderer]
+
+    template_name = 'ui/analytics.html'
+
+    def get(self, request, media_key):
+
+        response = super().get(request, media_key)
+
+        min_date = datetime.datetime.max
+        max_date = datetime.datetime.min
+        summed_by_date = {}
+
+        # Here we sum up all views for a particular day (irrespective of other variable) and
+        # calculate the min and max dates.
```

Comment:
I feel this is exactly the sort of thing the database should be doing...

Reply:
as discussed - will move this into client